recipe/conda.diff

```diff
@@ -0,0 +1,161 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 24ffd1225a..a5b68691fc 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -19,6 +19,18 @@ endif()
+ 
+ project(CATBOOST LANGUAGES C CXX ASM)
+ 
++# Patch to use system provided OpenSSL instead of vcpkg
++find_package(OpenSSL REQUIRED)
++include_directories(${OPENSSL_INCLUDE_DIR})
++add_library(LinkOpenSSL INTERFACE)
++target_link_libraries(LinkOpenSSL INTERFACE OpenSSL::SSL OpenSSL::Crypto)
++add_library(OpenSSL::OpenSSL ALIAS LinkOpenSSL)
```

Comment:
catboost assumes that `OpenSSL` comes from vcpkg and uses their cmake find files. This patch is to use the system provided instead of vcpkg.

Reply:
> catboost assumes that OpenSSL comes from vcpkg and uses their cmake find files.

No, catboost does not use vcpkg, catboost [uses OpenSSL from conan packages](https://github.com/catboost/catboost/blob/f903943a8cd903a117c3d3c8421cc72d3910562c/conanfile.py#L11).

>  This patch is to use the system provided instead of vcpkg.

Your patch above removes dependency on `conan` packages. CatBoost sub-libraries that depend on OpenSSL just have standard 

`find_package(OpenSSL)` syntax ( [example](https://github.com/catboost/catboost/blob/f903943a8cd903a117c3d3c8421cc72d3910562c/library/cpp/openssl/init/CMakeLists.linux-x86_64.txt#L15) ).

Without OpenSSL provided by conan they should be able to find system OpenSSL without any additional patches.

Reply:
the targets provided by `find_package(OpenSSL)` depends on if `FindOpenSSL.cmake` coming from the system or from conan.

1. conan supplied FindOpenSSL.cmake provides `OpenSSL::OpenSSL`
2. cmake supplied FindOpenSSL.cmake provides `OpenSSL::SSL` and `OpenSSL::Crypto`.

This is needed to reconcile the difference.

Reply:
Ok. Let's just add a comment in the file about it. Something like that:

```
# Use OpenSSL from system instead of the version provided by conan.
#  1. conan supplied FindOpenSSL.cmake provides OpenSSL::OpenSSL
#  2. cmake supplied FindOpenSSL.cmake provides OpenSSL::SSL and OpenSSL::Crypto.
# See https://github.com/conan-io/conan-center-index/issues/952
# This patch is needed to reconcile the difference.
```